clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1843,6 +1843,23 @@ static bool CheckAnyScalarOrVector(Sema *S, CallExpr *TheCall,
   return false;
 }
 
+static bool CheckNotBoolType(Sema *S, CallExpr *TheCall, unsigned ArgIndex) {
```

Comment:
Just forwarding this review comment here: https://github.com/llvm/llvm-project/pull/112400/files#r1837272356.

The reason I had made a helper function was because we will be using it in all the other `WaveActive...` ops. But I also don't know of any other use cases, so maybe we could just change to a `CheckWaveActiveType` type helper function instead?

Reply:
Addressed both these comments.